src/sentry/profiles/task.py

```diff
@@ -1109,19 +1109,35 @@ def determine_profile_type(profile: Profile) -> EventType:
     raise UnknownProfileTypeException
 
 
-def track_latest_sdk(project: Project, profile: Profile) -> None:
-    event_type = determine_profile_type(profile)
-
+def determine_client_sdk(profile: Profile, event_type: EventType) -> tuple[str, str]:
     client_sdk = profile.get("client_sdk")
 
-    if not client_sdk:
-        raise UnknownClientSDKException
+    if client_sdk:
+        sdk_name = client_sdk.get("name")
+        sdk_version = client_sdk.get("version")
+
+        if sdk_name and sdk_version:
+            return sdk_name, sdk_version
 
-    sdk_name = client_sdk.get("name")
-    sdk_version = client_sdk.get("version")
+    # some older sdks were sending the profile chunk without the
+    # sdk info, here we hard code a few and assign them a guaranteed
+    # outdated version
+    if event_type == EventType.PROFILE_CHUNK:
+        if profile["platform"] == "python":
+            return "sentry.python", "0.0.0"
+        elif profile["platform"] == "cocoa":
+            return "sentry.cocoa", "0.0.0"
+        elif profile["platform"] == "node":
+            # there are other node platforms but it's not straight forward
+            # to figure out which it is here so collapse them all into just node
```

Comment:
For Android we don't get this issue because the SDK was released recently and it set the `client_sdk` since the beginning, right?

Reply:
@stefanosiano do you have the answer to this?

Reply:
sdk name and versions are set in the envelope headers by default in the Android SDK since the beginning
v8.6.0 (ui profiling is officially out in 8.7.0) added platform in chunk header

Reply:
Can you elaborate on
> sdk name and versions are set in the envelope headers by default
Profiles are expecting the sdk info within the envelope item so if it's in the header, we'll need to handle that somehow.

And the difference between `v8.6.0` and `v8.7.0` is just that additional platform in the chunk header right?

Reply:
android envelopes (not the chunk item) contains `sdk` header, with name and version
```
{
  ...,
  "sdk": { 
    "name":"sentry.java.android",
    "version":"8.7.0",
    ...
  },
  ...
```

> And the difference between v8.6.0 and v8.7.0 is just that additional platform in the chunk header right?

`8.6.0` already has the platform in the chunk header. `8.7.0` moves all the ui profiling options out of experimental and implement last things - avoid stopping profiles immediately (it waits for current chunk to finish) and stops profile when app goes to background)

Reply:
> Okay, so
> 
> 1. we'll need to handle the sdk info coming from the header
> 2. the payload sent by 8.6.0 and 8.7.0 are identical

Ok, this something we'll handle in Relay, which means this PR is good to go?
Maybe I'd just add a comment in here explaining why for android we don't need to assign an outdated version.

Reply:
added in [3c8033a](https://github.com/getsentry/sentry/pull/89195/commits/3c8033acbf4da3ed3a85cbc9b11493e79c16d606)